### DISCUSSION TITLE
Simple list rendered as paragraph

### DISCUSSION THREAD

**Discussion by dgolda** [planet42/Laika#386] [General]
I have problem with writing lists.
When I have:
```
Foo
- bar
- baz
```

Laika treats it as one paragraph:
```
<p>
  Foo
  - bar
  - baz
</p>
```

So I have to write:
```
Foo

- bar

- baz
```
and have:
```
<p>
  Foo
</p>
<ul>
  <li>
    <p>
      bar
    </p>
  </li>
  <li>
    <p>
      baz
    </p>
  </li>
</ul>
```

But according to Common Mark spec it should work:
https://spec.commonmark.org/0.30/#example-303
https://spec.commonmark.org/dingus/?text=Foo%0A-%20bar%0A-%20baz%0A
```
Foo
- bar
- baz
```
gives:
```
<p>Foo</p>
<ul>
<li>bar</li>
<li>baz</li>
</ul>
```
Even on Github this list is rendered correctly as you can see:
Foo
- bar
- baz

What am I missing?

**Comment by jenshalm** [comment]
Ah, this is similar to #307. For now you can simply write it like this:

```markdown
Foo

-bar
-baz
```

You do not need the blank lines between list items.

When Laika integrates the full CommonMark test suite this will eventually work, too. But for now it's still mostly classic Markdown which was ambiguous about this.

My personal opinion is this: I would prefer the spec not to mandate this because I find that A) smashing together a paragraph and list without a blank line is less legible in Markdown, B) it slows down parsers unnecessarily as they have to look for new block elements in the middle of a paragraph, C) it makes the whole spec oddly inconsistent about where blank lines are necessary between two block items (which is most of the time, but then, sometimes, it's not) and D) this can easily trigger unintended list items, in particular with ordered/numbered lists when a sentence ending with a number and a period ends in the beginning of a line.

**Reply by dgolda** [reply]
Thank you. It's good enough for me.
